src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/LegacySyntaxNodeExtensions.cs

```diff
@@ -110,121 +135,156 @@ public static SyntaxNode LocateOwner(this SyntaxNode node, SourceChange change)
             return null;
         }
 
-        if (IsSpanKind(node))
+        if (node.IsSpanKind())
         {
             var editHandler = node.GetSpanContext()?.EditHandler ?? SpanEditHandler.CreateDefault();
             return editHandler.OwnsChange(node, change) ? node : null;
         }
 
-        IReadOnlyList<SyntaxNode> children;
-        if (node is MarkupStartTagSyntax startTag)
+        return node switch
         {
-            children = startTag.Children;
-        }
-        else if (node is MarkupEndTagSyntax endTag)
+            MarkupStartTagSyntax startTag => LocateOwnerForSyntaxList(startTag.Children, change),
+            MarkupEndTagSyntax endTag => LocateOwnerForSyntaxList(endTag.Children, change),
+            MarkupTagHelperStartTagSyntax startTagHelper => LocateOwnerForSyntaxList(startTagHelper.Children, change),
+            MarkupTagHelperEndTagSyntax endTagHelper => LocateOwnerForSyntaxList(endTagHelper.Children, change),
+            _ => LocateOwnerForChildSyntaxList(node.ChildNodes(), change)
+        };
+
+        static SyntaxNode? LocateOwnerForSyntaxList(in SyntaxList<RazorSyntaxNode> list, SourceChange change)
         {
-            children = endTag.Children;
-        }
-        else if (node is MarkupTagHelperStartTagSyntax startTagHelper)
-        {
-            children = startTagHelper.Children;
-        }
-        else if (node is MarkupTagHelperEndTagSyntax endTagHelper)
-        {
-            children = endTagHelper.Children;
-        }
-        else
-        {
-            children = node.ChildNodes();
+            foreach (var child in list)
+            {
+                if (child.LocateOwner(change) is { } owner)
+                {
+                    return owner;
+                }
+            }
+
+            return null;
         }
 
-        SyntaxNode owner = null;
-        for (int i = 0; i < children.Count; i++)
+        static SyntaxNode? LocateOwnerForChildSyntaxList(in ChildSyntaxList list, SourceChange change)
         {
-            var child = children[i];
-            owner = LocateOwner(child, change);
-            if (owner != null)
+            foreach (var child in list)
             {
-                break;
+                if (child.LocateOwner(change) is { } owner)
+                {
+                    return owner;
+                }
             }
-        }
 
-        return owner;
+            return null;
+        }
     }
 
     public static bool IsTransitionSpanKind(this SyntaxNode node)
     {
-        if (node == null)
+        if (node is null)
```

Comment:
Are all of these `==` -> `is` refactorings necessary?

Reply:
Probably not. I just wanted to make sure the file is consistent.